container/go/pkg/compat/ociToLegacy.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2015 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//////////////////////////////////////////////////////////////////////
+// This utility works with the new_container_pull and new_container_load targets
+// to generate the appropriate pseudo-intermediate format that is compatible
+// with container_import rule.
+
+package compat
+
+import (
+	ospkg "os"
+	"path"
+	"strconv"
+
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/pkg/errors"
+)
+
+// Where the pulled OCI image artifacts are stored in.
+const artifactsDir = "image-oci"
+
+// Where the alias to the correct config.json and layers.tar.gz are found
+const symlinksDir = "image"
+
+// Extension for layers and config files that are made symlinks
+const targz = ".tar.gz"
+const configExt = "config.json"
+
+// generateSym creates predictable symbolic links to the config.json and layer .tar.gz files
+// so that they may be easily consumed by container_import targets.
+// The dstPath is the top level directory in which the puller will create symlinks inside an image/ directory
+// pointing to actual pulled OCI image artifacts in image-oci/ directory.
+func generateSymlinks(img v1.Image, dstPath string) error {
```

Comment:
call this LegacyFromOCIImage. I think the symlink creation should remain an implementation detail of this package.